File: Book.cs
using System.Collections.Generic;

namespace BookModel
{
	public class Book : IBookElement
	{
		public string Title { get; set; }
		private List<Author> _authors;
		private List<IBookElement> _children;

		public Book(string title)
		{
			Title = title;
			_authors = new List<Author>();
			_children = new List<IBookElement>();
		}

		public void AddAuthor(Author author)
		{
			_authors.Add(author);
		}

		public void Print()
		{
			Console.WriteLine($"Book: {Title}");
			if (_authors.Count > 0)
			{
				Console.WriteLine("Authors:");
				foreach (var a in _authors) a.Print();
			}
			foreach (var child in _children) child.Print();
		}

		public void Add(IBookElement element)
		{
			_children.Add(element);
		}

		public void Remove(IBookElement element)
		{
			_children.Remove(element);
		}

		public IBookElement Get(int index)
		{
			return _children[index];
		}

		public void AddContent(IBookElement element)
		{
			Add(element);
		}

		public override string ToString()
		{
			return $"Book: {Title}";
		}
	}
}

File: src/main/java/Book.java
import java.util.ArrayList;
import java.util.List;

public class Book implements Element {
    private String title;
    private List<Author> authors = new ArrayList<>();
    private List<Element> content = new ArrayList<>();

    public Book(String title) {
        this.title = title;
    }

    public void addAuthor(Author author) {
        this.authors.add(author);
    }
    
    public void addContent(Element element) {
        this.content.add(element);
    }

    @Override
    public void print() {
        System.out.println("Book: " + title);
        
        System.out.println("Authors:");
        for (Author author : authors) {
            author.print();
        }

        for (Element element : content) {
            element.print();
        }
    }
    

    @Override
    public void add(Element element) {
        addContent(element);
    }
    @Override
    public void remove(Element element) {
        content.remove(element);
    }
    @Override
    public Element get(int index) {
        return content.get(index);
    }
}